datafusion/functions/src/datetime/to_char.rs

```diff
@@ -215,8 +219,19 @@ fn _to_char_array(args: &[ColumnarValue]) -> Result<ColumnarValue> {
         };
         // this isn't ideal but this can't use ValueFormatter as it isn't independent
         // from ArrayFormatter
-        let formatter = ArrayFormatter::try_new(arrays[0].as_ref(), &format_options)?;
-        let result = formatter.value(idx).try_to_string();
+        let result = match format {
+            Some(_) => {
```

Comment:
I think in this case we should also be formatting to None, as in change
```rust
let mut results: Vec<String> = vec![];
```

to 
```rust
let mut results: Vec<Option<String>> = vec![];
```
so that it can properly represent nulls

Reply:
I'm using the `String` type only. Reason is same as https://github.com/apache/arrow-datafusion/pull/9689/files#r1532580061 .

Reply:
Updated:
Type has been changed to `Vec<Option<String>>` now.